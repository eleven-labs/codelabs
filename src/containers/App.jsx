import React from 'react';
import { connect } from 'react-redux';
import { renderRoutes } from 'react-router-config';

import Navigation from '../components/Navigation';
import { addTopic } from '../actions';

const mapStateToProps = ({ courses, topics }) => ({ courses, topics });
const mapDispatchToProps = { addTopic };

export class CodeLabApp extends React.Component {
  render() {
    const { courses, topics, route } = this.props;

    return (
      <div>
        <Navigation />
        {route && renderRoutes(route.routes, { topics, courses })}
      </div>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(CodeLabApp);
