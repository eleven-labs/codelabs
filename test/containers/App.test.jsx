import 'babel-polyfill';
import React from 'react';
import { MemoryRouter } from 'react-router';
import { mount } from 'enzyme';

import App from '../../src/containers/App';

describe('<App />', () => {
  it('should render correctly', () => {
    const loc = { pathname: '/', query: { a: 'b' } }

    const wrapper = mount(
      <MemoryRouter initialEntries={[loc]} initialIndex={0}>
        <App />
      </MemoryRouter>
    );

    expect(wrapper.find('Header')).toHaveLength(1);
    expect(wrapper.find('Search')).toHaveLength(1);
  });
});
