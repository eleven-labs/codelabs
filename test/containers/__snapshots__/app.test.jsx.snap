// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav>
    <a
      href="/"
      onClick={[Function]}
    >
      home
    </a>
     
    <a
      href="/list"
      onClick={[Function]}
    >
      list
    </a>
  </nav>
</div>
`;
